src/catalog/src/lib.rs

```diff
@@ -44,26 +46,52 @@ pub trait CatalogManager: Send + Sync {
 
     async fn catalog_names(&self) -> Result<Vec<String>>;
 
-    async fn schema_names(&self, catalog: &str) -> Result<Vec<String>>;
+    async fn schema_names(&self, catalog: &str, channel: Channel) -> Result<Vec<String>>;
```

Comment:
Can we use `Option<QueryContextRef>` for all `CatalogManager` functions? I think it makes more sense to provide whole context instead of just `channel`. It make behaviour of `CatalogManager` more extensible.